lib/parser.js

```diff
@@ -190,13 +190,23 @@ async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileForm
 
   const schemaFormat = msg.schemaFormat || DEFAULT_SCHEMA_FORMAT;
 
-  await PARSERS[String(schemaFormat)]({
+  const parser = PARSERS[String(schemaFormat)];
+
+  if (parser === undefined) {
+    throw new ParserError({
+      type: 'schema-parser-not-registered',
+      title: 'The file does not correspond to a supported schema format',
+      detail: 'The asyncApi document cannot be handled by provided parsers.\nIt occurred while trying to parse the following' +
```

Comment:
```suggestion
      detail: `The following message payload "${pathToPayload}" has "${schemaFormat}" schema format which isn't supported by parser instance. Add a missed format parser through the "registerSchemaParser" function.`
```

Reply:
I'm not sure if we should also give the last note regarding the use of the function @derberg

Reply:
I like such detailed errors, also would be cool to refer specific implementations if known